iroh/src/util/fs.rs

```diff
@@ -171,11 +172,70 @@ pub async fn load_secret_key(key_path: PathBuf) -> anyhow::Result<SecretKey> {
     }
 }
 
+/// Information about the content on a path
+#[derive(Debug, Clone)]
+pub struct PathContent {
+    /// total size of all the files in the directory
+    pub size: u64,
+    /// total number of files in the directory
+    pub files: u64,
+}
+
+/// Walks the directory to get the total size and number of files in directory or file
+pub fn path_content_info(path: impl AsRef<Path>) -> anyhow::Result<PathContent> {
+    path_content_info0(path)
+}
+
+fn path_content_info0(path: impl AsRef<Path>) -> anyhow::Result<PathContent> {
+    let mut files = 0;
+    let mut size = 0;
+    let path = path.as_ref();
+
+    if path.is_dir() {
+        for entry in read_dir(&path).context("read_dir err")? {
```

Comment:
https://crates.io/crates/walkdir is probably a good deal faster

Reply:
switched to `walkdir` & got a stack overflow when adding a dir w/ 30k entries, so switched it back. lmk if you have an insight.

Reply:
Oo, I would definitely file an issue, that is surprising